apps/folders/watchers/watcher_folder.go

```diff
@@ -0,0 +1,63 @@
+package watchers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/grafana/grafana-app-sdk/resource"
+	folder "github.com/grafana/grafana/pkg/apis/folder/v0alpha1"
+	grafanarest "github.com/grafana/grafana/pkg/apiserver/rest"
+	"k8s.io/apimachinery/pkg/apis/meta/internalversion"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog/v2"
+)
+
+type FolderWatcher struct {
+	storage grafanarest.Storage
+}
+
+func NewFolderWatcher(s grafanarest.Storage) (*FolderWatcher, error) {
+	return &FolderWatcher{
+		storage: s,
+	}, nil
+}
+
+// Delete handles delete events for folder.Folder resources.
+func (w *FolderWatcher) Delete(ctx context.Context, rObj resource.Object) error {
+	metadata := rObj.GetStaticMetadata()
+	if metadata.Kind != folder.RESOURCE {
+		return fmt.Errorf("provided object is not of type *folder.Folder (name=%s, namespace=%s, kind=%s)",
+			metadata.Name, metadata.Namespace, metadata.Kind)
+	}
+
+	// find all children of the object that just got deleted
+	parent := rObj.GetName()
+	namespace := rObj.GetNamespace()
+	matched, err := w.storage.List(ctx, &internalversion.ListOptions{TypeMeta: meta.TypeMeta{Kind: folder.RESOURCE}, FieldSelector: fields.AndSelectors(
+		fields.OneTermEqualSelector("metadata.annotations.grafana.app/folder", parent),
+		fields.OneTermEqualSelector("metadata.namespace", namespace),
+	)})
+	if err != nil {
+		klog.InfoS("could not list resources", "error", err)
+	}
+
+	obj, ok := matched.(resource.Object)
```

Comment:
`obj` will be a list, therefore we should iterate over each item, no?

Reply:
this is what I thought! I'm also finding it weird that `List` (which fulfils the [rest k8s interface](https://github.com/kubernetes/apiserver/blob/master/pkg/registry/rest/rest.go#L130)) doesn't return a listObject but rather a runtime one 🤷‍♀️